src/wagi_app.rs

```diff
@@ -289,12 +320,38 @@ fn parse_handler_configuration_source(
                 ))
             }
         }
-        // Case: got a bindle id and server URL. Can't have a bindir dir or module file.
-        (Some(bindle_id), None, Some(bindle_url), None) => {
+        // Cases: got a bindle id and server URL. Can't have a bindir dir or module file.
+        // No basic http auth creds for bindle server supplied
+        (Some(bindle_id), None, Some(bindle_url), None, None, None) => {
```

Comment:
This duplicates a considerable amount of code and breaks up the flow of cases even more than it was already broken up (which I admit was quite a bit).

An alternative to consider:

* Keep the match pattern as it is
* Write a helper method that matches on the user and password values to create an auth object
* Use that auth object to create the Bindle client

The perennial problem with this, I know, and something that really vexes me about the Bindle client design, is the auth type being a type-level parameter of the client.  The solution is to nick the `AnyAuth` type from the Hippo CLI and use that as the auth type.

This should be how the Bindle client works by default and I will die on this hill.

Reply:
Thanks @itowlson.  I pushed a commit which refactors things with an approach derived from hippo-cli.  I do like how it slims down the CLI config -> emplacing a remote bindle flow, I'm anticipating there are still some things to tweak.  One: I couldn't figure out how to keep the 'Debug' trait on the structs that use the newly introduced BindleConnectionInfo struct/impl due to the underlying token_manager, so I removed the trait to get things working -- no doubt this isn't ideal 😂.  WDYT?